tree/forest/v7/inc/ROOT/RColumnModel.hxx

```diff
@@ -0,0 +1,60 @@
+/// \file ROOT/RColumnModel.hxx
+/// \ingroup Forest ROOT7
+/// \author Jakob Blomer <jblomer@cern.ch>
+/// \date 2018-10-09
+/// \warning This is part of the ROOT 7 prototype! It will change without notice. It might trigger earthquakes. Feedback
+/// is welcome!
+
+/*************************************************************************
+ * Copyright (C) 1995-2015, Rene Brun and Fons Rademakers.               *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+#ifndef ROOT7_RColumnModel
+#define ROOT7_RColumnModel
+
+#include <ROOT/RStringView.hxx>
+
+namespace ROOT {
+namespace Experimental {
+
+// clang-format off
+/**
+\class ROOT::Experimental::EColumnType
+\ingroup Forest
+\brief The possible trivial, native content types of a column
+*/
+// clang-format on
+enum class EColumnType {
+   kUnknown,
+   kIndex, // type for root columns of nested collections
+   kByte,
+   kReal64,
+   kReal32,
+   kReal16,
+   kReal8,
+   kInt64,
+   kInt32,
+   kInt16,
+   //...
+};
+
+// clang-format off
+/**
+\class ROOT::Experimental::RColumnModel
+\ingroup Forest
+\brief Holds the static meta-data of a column in a tree
+*/
+// clang-format on
+class RColumnModel {
+public:
+   RColumnModel(std::string_view name, EColumnType type, bool is_sorted);
```

Comment:
It's unclear to me what `is_sorted` means. What is the difference between a sorted and unsorted column model?

Reply:
I'll remove it for time being, it is not essential at this point. Eventually we should store wether columns have values in sorted order because it is easy to store and verify on `Fill()` and we can use it to do binary search on such columns.  For instance, on a column that stores time stamps.